test/src/test/java/org/corfudb/integration/ClusterReconfigIT.java

```diff
@@ -599,4 +600,138 @@ public void testSystemDownHandler() throws Exception {
         // Wait for the systemDownHandler to be invoked.
         semaphore.tryAcquire(PARAMETERS.TIMEOUT_LONG.toMillis(), TimeUnit.MILLISECONDS);
     }
+
+    /**
+     * Ensure that multiple resets on the same epoch reset the log unit server only once.
+     * Consider 3 nodes 9000, 9001 and 9002.
+     * Kill node 9002 and recover it so that the heal node workflow is triggered, the node is
+     * reset and added back to the chain.
+     * We then perform a write on addresses 5-9.
+     * Now we attempt to reset the node 9002 again on the same epoch and assert that the write
+     * performed on address 1 is not erased by the new reset.
+     */
+    @Test
+    public void preventMultipleResets() throws Exception {
+        // Set up cluster of 3 nodes.
```

Comment:
This test will pass even if you comment out batchWriter.reset() in resetLogUnit, I think It will be a better test to check that all of the data exists. The reason it passes is because the entries are holes, so isEmpty() is returning false.